---
title: "Proposal"
author: "Jaden Anderson"
date: "April 12, 2018"
output: html_document

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(knitr)
library(seqinr)
```

# Introduction

Selection pressures can be analyzed in a variety of fashions. Many studies have been done studying genetic drift of ecological populations or positive/negative selection on specific genes of interest in order to understand their evolutionary history and progression. Here I propose a unique and complicated case of evolution: Outer membrane beta barrels (OMBBs). 

Gram negative bacteria have a unique arrangement of membranes that help them persist in their enviornment: an outer and inner membrane. On these membranes exist a variety of proteins and structures that have a range of functions that the bacteria utilize. Outer membrane proteins are used to interact with their extracellular enviornment that have a diverse array of functions like import and export of substances, cell adhesion, proteolysis, membrane protein insertion, and more. Even though proteins on the outer membrane have a variety of functions, surprisingly the secondary structure of these proteins (i.e. the structure of what the proteins look like) are very similar. While these secondary structures are highly similar, their sequence composition (or similarity) is highly variable which makes studying their evolution very difficult. The most notable difference between the OMBBs is the number of strands which has illuminated a unique conservation pattern, most notiably seen in the 16 stranded barrels where the N-terminal half of the protein is unconserved and varies greatly and the C-terminal half is highly conserved and varies little.![_Figure 1:_ This figure is taken from "The Complex Evolution of Outer Membrane Proteins" which is sited in this proposal. This figure illustrates the unique conservation pattern found by the authors. Here when we look at conserved strand identity, we can see a low level of conservation through strands 1-8 and an increase in strand conservation in in strands 9-16. This interesting conservation pattern may be due to a unique combination of selection pressures on the 16 stranded barrels.](c_term_conservation.png) Using this data, my hyposthesis is that the 16 stranded OMBBs may under go multiple selection patterns which would be an interesting occurance that would warrant more study. 

# Proposed Methods
Looking at patterns of conservation, we can start to address some possible directions for selection pressures. In order to determine if a protein is undergoing positive or negative selection we can calculate the ratio of nonsynonmous mutations (amino acid changes at that position) to synonymous mutations (amino acid does not change at that position) or better known as dN/dS ratios (or Ka/Ks). There exists a package in R called seqinr and in this package exists a function called kaks that will calculated dN/dS ratios. The rough formula for calculating this ratio is $\omega = dN/dS \approx (selection + neutral)/neutral $ where selection is roughly equal to the average number of substituions in aligned sequences and neutral is the evolutionary distance between the sequences ignoring selection (i.e. synonymous mutations). If dN/dS is around 1, selection has not occured on the sequences and there is no "selection pressures" by these means, if it is lower than 1 there is negative selection or changes in the sequences are selected against (conservation of sequence) and if it is higher than 1 there is positive selection or changes in the sequences are selected for. Using these ratios we can try to piece together the selection pressures on the 16 stranded OMBBs. One possibility is that the N-terminal half (strands 1-8) undergo positive selection as their sequence conservation is low and the C-terminal half (strands 9-16) undergo negative selection as their sequence conservation is high. The overal hypotheses I have are outlined in table 1.

--------------------------------------
```{r, hypothesis_table, echo = FALSE}

hypothesis_table <- read.table("hypothesis_table.txt", sep = "\t", header = TRUE)

kable(hypothesis_table, caption = "_Table 1._ This table outlines the main selection pressure hypotheses I have for this system. N-terminal and C-terminal denote the position of the protein of interest, and positive, negative and none indicate the type of selection pressure. ")
```
--------------------------------------